### PR TITLE
Adds our hiring process

### PR DESCRIPTION
**Pull request by jtslear**
* Removes the lingering sentence about the dev advocacy position
* Updates the blurb about our stack
* connect #3581

Signed-off-by: John T Skarbek <jtslear@gmail.com>

### CODE REVIEW

File: HIRING_PROCESS.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+### Cover letter and resume
+Start your conversation with us by sending your cover letter and resume to
+[team@waffle.io](mailto:team@waffle.io).  In the cover letter, include why you
+want to work at Waffle and what you believe you bring to the team.
+
+This is important to us for trying to assess both the experience and level of
+passion relevant to the position.
+
+After we take a gander, if we think are a good fit, or potentially in doubt,
+we'll reach out to you to start the screening.
```
Comment (BAMason) [inline HIRING_PROCESS.md]: This sentence reads a little weird to me.  Should it be `if we think *you* are a good fit` or `if we think it's a good fit`? and if there were potential doubts, would we still reach out to start screening?

What if this just said: `After we take a gander and if it looks like a good fit, we'll reach out to start the screening process.`

File: HIRING_PROCESS.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+such that we can gauge your relevant experience.  This should only take a couple
+of hours. It is up to you to decide what to build. We are looking for creativity
+and ability.  As mentioned above if we had doubts about moving you forward, this
+is your opportunity to impress us.
+
+### Interview
+Our next step is an on-site interview.  Here you'll meet the team and vice
+versa. You'll be interviewed on technical competency and cultural fit. No live
+coding will occur. We will also discuss compensation/benefits at this time. Be
+sure to checkout what we are looking for on our [Careers page](./CAREERS.md)
```
Comment (BAMason) [inline HIRING_PROCESS.md]: Should we bump this `checkout what we are looking for` piece up to the process section where we talk about fit?

File: HIRING_PROCESS.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+### Coding exercise
+After the interview, we'll setup a domain specific coding challenge.  We ask
+that you don't spend more than 10 hours to work through it and you given a full 7
```
Comment (AnaMSteele) [inline HIRING_PROCESS.md]: correction for "and you given" to  "and you'll be given"

File: HIRING_PROCESS.md (hunk 2 above)
Comment (AnaMSteele) [reply HIRING_PROCESS.md]: 👍

File: HIRING_PROCESS.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+want to work at Waffle and what you believe you bring to the team.
+
+This is important to us for trying to assess both the experience and level of
+passion relevant to the position.
+
+After we take a gander, if we think are a good fit, or potentially in doubt,
+we'll reach out to you to start the screening.
+
+### Screening
+At this point we want to learn more about you, your history, and your future
```
Comment (AnaMSteele) [inline HIRING_PROCESS.md]: suggested rewrite strike "At this point" start sentence "We want to learn"

File: HIRING_PROCESS.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+professional, academic, or personal projects.
+1. What is a technology or technique that you have recently learned and enjoyed.
+Tell us why we should learn it.
+
+The second item we'll send you is a webtask.
+
+Using [webtask.io](https://webtask.io) we'll ask you to put together a small app
+such that we can gauge your relevant experience.  This should only take a couple
+of hours. It is up to you to decide what to build. We are looking for creativity
+and ability.  As mentioned above if we had doubts about moving you forward, this
```
Comment (AnaMSteele) [inline HIRING_PROCESS.md]: If you strike from the Cover Letter Resume section "or potentially in doubt" Then also update this sentence to say "This is your opportunity to impress us."

**Pull request merged**